src/main/kotlin/sliv/tool/catalogue/view/CatalogueFileNamesFieldsView.kt

```diff
@@ -0,0 +1,114 @@
+package sliv.tool.catalogue.view
+
+import javafx.collections.ObservableList
+import javafx.scene.Scene
+import javafx.scene.control.Labeled
+import javafx.scene.control.SelectionMode
+import javafx.scene.image.Image
+import javafx.scene.image.WritableImage
+import javafx.scene.input.ClipboardContent
+import javafx.scene.input.DragEvent
+import javafx.scene.input.TransferMode
+import sliv.tool.catalogue.*
+import sliv.tool.catalogue.controller.CatalogueController
+import sliv.tool.catalogue.model.CatalogueField
+import sliv.tool.project.model.ProjectFrame
+import sliv.tool.scene.view.SceneView
+import tornadofx.*
+import kotlin.math.min
+
+class CatalogueFileNamesFieldsView(fields: ObservableList<CatalogueField>): View(), SelectionNode {
+    companion object {
+        private const val ListViewFieldCellHeight = 30.0
+        private const val ListViewFieldIconSize = 20.0
+
+        private const val DragViewMaxFieldsNumber = 100
+    }
+
+    private val controller: CatalogueController by inject()
+    private val sceneView: SceneView by inject()
+
+    override val areSelectedAllItems: Boolean
+        get() = fileNamesListView.selectedItemsCount == controller.model.frames.count()
+    override val isSelectionEmpty: Boolean
+        get() = fileNamesListView.selectedItems.isEmpty()
+    override val selectedItems: List<CatalogueField>
+        get() = fileNamesListView.selectedItems
+    override val selectedFrames: List<ProjectFrame>
+        get() = fileNamesListView.selectedItems.map { it.frame }
+
+    private var isDragging = false
+
+    private val fileNamesFieldIconImage =
+        Image(this.javaClass.classLoader.getResource("catalogue_image_icon.png")?.openStream())
+
+    private fun initializeDragEvents() {
+        fileNamesListView.setOnDragDetected {
+            onCatalogueDragDetected()
+        }
+        sceneView.root.addEventFilter(DragEvent.DRAG_OVER, ::onSceneDragOver)
+        sceneView.root.addEventFilter(DragEvent.DRAG_DROPPED, ::onSceneDragDropped)
+    }
+
+    private fun setFileNamesListViewCellFormat(labeled: Labeled, item: CatalogueField?) {
+        labeled.text = item?.fileName
+        labeled.graphic = imageview(fileNamesFieldIconImage) {
+            fitHeight = ListViewFieldIconSize
+            isPreserveRatio = true
+        }
+        labeled.prefHeight = ListViewFieldCellHeight
+    }
+
+    private val fileNamesListView = listview(fields) {
+        selectionModel.selectionMode = SelectionMode.MULTIPLE
+
+        cellFormat {
+            setFileNamesListViewCellFormat(this, it)
+        }
+    }
+
+    override val root = fileNamesListView.also { initializeDragEvents() }
+
+    private fun onSceneDragDropped(event: DragEvent) {
```

Comment:
this should be in separate delegate to prevent knowledge about scene view from catalogue view

Reply:
maybe done as separate issue